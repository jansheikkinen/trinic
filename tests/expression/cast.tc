// Cast

function main(void) -> void do
  x as float32; y as struct TestStruct; z as *[][]struct TestStructB(int8);
end
