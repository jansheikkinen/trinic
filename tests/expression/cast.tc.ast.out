## PARSER ##
Parsing tests/expression/cast.tc...

### LEXER ###
Generating tokens from source...
Generated 31 tokens
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:   IDENTIFIER,          x, (004, 003)
009:           AS,     (null), (004, 006)
010:      FLOAT32,     (null), (004, 014)
011:            ;,     (null), (004, 015)
012:   IDENTIFIER,          y, (004, 017)
013:           AS,     (null), (004, 020)
014:       STRUCT,     (null), (004, 027)
015:   IDENTIFIER, TestStruct, (004, 038)
016:            ;,     (null), (004, 039)
017:   IDENTIFIER,          z, (004, 041)
018:           AS,     (null), (004, 044)
019:            *,     (null), (004, 046)
020:            [,     (null), (004, 047)
021:            ],     (null), (004, 048)
022:            [,     (null), (004, 049)
023:            ],     (null), (004, 050)
024:       STRUCT,     (null), (004, 056)
025:   IDENTIFIER, TestStructB, (004, 068)
026:            (,     (null), (004, 069)
027:         INT8,     (null), (004, 073)
028:            ),     (null), (004, 074)
029:            ;,     (null), (004, 075)
030:          END,     (null), (005, 003)

### AST GENERATION ###
Declaration FUNCTION  main
  Stmt EXPRESSION
    Expr CAST
      Expr VARIABLE x
      Type TYPE FLOAT32
  Stmt EXPRESSION
    Expr CAST
      Expr VARIABLE y
      Type STRUCTURE TestStruct
  Stmt EXPRESSION
    Expr CAST
      Expr VARIABLE z
      Type POINTER
        Type ARRAY
          Type ARRAY
            Type STRUCTURE TestStructB
              Arg SUMARGTYPE
                Sum_Arg_Type 
                  Type TYPE INT8
